### PR TITLE
add a sessions smoke test to examples

### PR DESCRIPTION
**Pull request by hppritcha**
Demonstrate the essential feature of sessions, the ability to start/stop and also interop with MPI Init/Finalize multiple times.

On to the future.

Signed-off-by: Howard Pritchard <howardp@lanl.gov>

### CODE REVIEW

**Review by awlauria — APPROVED** [review]

File: examples/hello_sessions.c [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    rc = MPI_Session_init (MPI_INFO_NULL, MPI_ERRORS_RETURN, &session2);
+    if (MPI_SUCCESS != rc) {
+        print_error("Third session initialization failed", rc);
+        return -1;
+    }
+
+    rc = MPI_Session_finalize (&session2);
+    if (MPI_SUCCESS != rc) {
+        print_error("Second session finalize failed", rc);
```
Comment (awlauria) [inline examples/hello_sessions.c:49]: Shouldn't this read 'first'?

**Comment by jsquyres** [comment]
Before this is merged, can we talk about the purpose of the `examples` directory?

File: examples/hello_sessions.c [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        return -1;
+    }
+
+    rc = MPI_Session_finalize (&session);
+    if (MPI_SUCCESS != rc) {
+        print_error("Third session finalize failed", rc);
```
Comment (awlauria) [inline examples/hello_sessions.c:55]: 'second' here?

**Comment by awlauria** [comment]
I do think having this in the examples directory is a good idea, regardless.

**Comment by jsquyres** [comment]
> I do think having this in the examples directory is a good idea, regardless.

Is it Open MPI's responsibility to educate users on new MPI-4 things?

**Comment by awlauria** [comment]
> > I do think having this in the examples directory is a good idea, regardless.
> 
> Is it Open MPI's responsibility to educate users on new MPI-4 things?

No - but I really don't see a downside.

**Comment by jsquyres** [comment]
> No - but I really don't see a downside.

The downside is: where do we stop, and how do we explain it to customers?  For example, do we need to put in smoke tests for all MPI functions here?  Having hello world + ring is one thing, having a set of tests is another -- which tests should be included?  Which should not?  Why have some tests and not others?  ... etc.

**Pull request closed without merging**